### DISCUSSION TITLE
how to extend CORS or Allow-Oigin list in browser project

### DISCUSSION THREAD

**Discussion by mate1213** [AvaloniaUI/Avalonia#16522] [Q&A] [answered]
I want to try build an app and all of my queries to the server shut downs by the CORS rule (which how it should be now).
I want to add my backend URL to the allow origin list, but I cannot find a way to do it. As far as I understand some visualStudio settings should be changed when the FrontEnd page served it's filled correctly.
I've found a ASP.NET way to it, but in avalonia, I think that will be not sufficient. (I'm not really know how WASM works, didn't dig deep into it until now.)

**Comment by mate1213** [comment, marked as answer]
Figured out:
In launchSettings.json you can put:
`"AllowedHosts": "*"` or `"AllowedHosts": "https://domain1.com,https://domain2.com"`

**Reply by nkbooth** [reply]
I'm fairly certain that I'm losing my mind, but this is exactly the problem I'm trying to solve today, except adding an AllowedHosts list continues to give me fits.  Can you share the documentation you found?

**Reply by mate1213** [reply]
You have to follow ASP.NETCore Kestrel server documentation.
Link: https://learn.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/endpoints?view=aspnetcore-8.0#configure-endpoints-in-appsettingsjson
Example:
```
{
  "profiles": {
    "AvaloniaApplication2.Browser": {
      "commandName": "Project",
      "launchBrowser": true,
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development",
      },
      "applicationUrl": "https://localhost:7169;http://localhost:5235",
      "inspectUri": "{wsProtocol}://{url.hostname}:{url.port}/_framework/debug/ws-proxy?browser={browserInspectUri}",
      "AllowedHosts": "*"
    }
}
```
